From b74a61ee6fe1ad8520d1c9abb5663f8afcf7c65f Mon Sep 17 00:00:00 2001
From: Aravind Mosam <amosam@qti.qualcomm.com>
Date: Mon, 7 Oct 2024 01:46:59 +0530
Subject: [PATCH] hostpad: Upgrade package to 13th Sept 2024

Upgrade the hostap package to version ccba6921de63, dated 13th Sept 2024.

Signed-off-by: Aravind Mosam <amosam@qti.qualcomm.com>
---
 package/network/services/hostapd/Makefile | 101 +++++++++++++---------
 1 file changed, 59 insertions(+), 42 deletions(-)

diff --git a/package/network/services/hostapd/Makefile b/package/network/services/hostapd/Makefile
index b078aee406c0..07cb08bb30e0 100644
--- a/package/network/services/hostapd/Makefile
+++ b/package/network/services/hostapd/Makefile
@@ -9,20 +9,13 @@ PKG_RELEASE:=2
 
 PKG_SOURCE_URL:=https://w1.fi/hostap.git
 PKG_SOURCE_PROTO:=git
-PKG_SOURCE_DATE:=2024-09-15
-PKG_SOURCE_VERSION:=5ace39b0a4cdbe18ddbc4e18f80ee3876233c20b
-PKG_MIRROR_HASH:=43187d1ddd923970ef3821a0fe267880c2483f6f1a5acd0dd2d7906661ab1372
-
-EXTERNAL_MAKEFILE:=$(TOPDIR)/openwrt-patches/package/network/services/hostapd
-
-ifneq (,$(findstring QSDK_Open,$(CONFIG_TARGET_PROFILE)))
-	include $(EXTERNAL_MAKEFILE)/hostapd.mk
-endif
-PKG_SRC_DIR:=$(EXTERNAL_MAKEFILE)
-
+PKG_SOURCE_DATE:=2024-09-13
+PKG_SOURCE_VERSION:=ccba6921de6372a2220350bb5ed5776ea8c76bbb
+PKG_MIRROR_HASH:=e6ccdc31646e96c663d4948c897eddd0b1e73dd6af50686a6108e064073679e2
 PKG_MAINTAINER:=Felix Fietkau <nbd@nbd.name>
 PKG_LICENSE:=BSD-3-Clause
 PKG_CPE_ID:=cpe:/a:w1.fi:hostapd
+PKG_SOURCE_SUBMODULES:=skip
 
 PKG_BUILD_PARALLEL:=1
 PKG_ASLR_PIE_REGULAR:=1
@@ -104,7 +97,7 @@ DRIVER_MAKEOPTS= \
 	CONFIG_IEEE80211AX=$(HOSTAPD_IEEE80211AX) \
 	CONFIG_IEEE80211BE=$(HOSTAPD_IEEE80211BE) \
 	CONFIG_MBO=$(CONFIG_WPA_MBO_SUPPORT) \
-	CONFIG_UCODE=y CONFIG_APUP=y
+	CONFIG_UCODE=y
 
 ifeq ($(SSL_VARIANT),openssl)
   DRIVER_MAKEOPTS += CONFIG_TLS=openssl CONFIG_SAE=y
@@ -194,6 +187,16 @@ endef
 
 Package/hostapd-openssl/description = $(Package/hostapd/description)
 
+define Package/hostapd-macsec
+$(call Package/hostapd/Default,$(1))
+  TITLE+= (macsec)
+  VARIANT:=macsec
+endef
+
+define Package/hostapd-macsec/description
+This package is Hostapd Authenticator for macsec support.
+endef
+
 define Package/hostapd-wolfssl
 $(call Package/hostapd/Default,$(1))
   TITLE+= (wolfSSL full)
@@ -203,16 +206,6 @@ endef
 
 Package/hostapd-wolfssl/description = $(Package/hostapd/description)
 
-define Package/hostapd-macsec
-$(call Package/hostapd/Default,$(1))
-  TITLE+= (macsec)
-  VARIANT:=macsec
-  DEPENDS+=@TARGET_ipq95xx || TARGET_ipq53xx || TARGET_ipq54xx \
-	   +kmod-qca-nss-macsec $(OPENSSL_DEPENDS)
-endef
-
-Package/hostapd-macsec/description = $(Package/hostapd/description)
-
 define Package/hostapd-mbedtls
 $(call Package/hostapd/Default,$(1))
   TITLE+= (mbedTLS full)
@@ -465,6 +458,16 @@ define Package/wpa-supplicant/config
 	source "$(SOURCE)/Config.in"
 endef
 
+define Package/wpa-supplicant-macsec
+$(call Package/wpa-supplicant/Default,$(1))
+  TITLE:=WPA Supplicant (MACSEC)
+  VARIANT:=supplicant-macsec
+endef
+
+define Package/wpa-supplicant-macsec/Description
+  WPA Supplicant with MACSEC support.
+endef
+
 define Package/wpa-supplicant-p2p
 $(call Package/wpa-supplicant/Default,$(1))
   TITLE+= (Wi-Fi P2P support)
@@ -583,12 +586,6 @@ define Package/eapol-test-mbedtls
   PROVIDES:=eapol-test
 endef
 
-ifneq (,$(findstring QSDK_Open,$(CONFIG_TARGET_PROFILE)))
-  define Build/Prepare
-	$(CP_SRC_DIR)
-  endef
-endif
-
 ifneq ($(wildcard $(PKG_BUILD_DIR)/.config_*),$(subst .configured_,.config_,$(STAMP_CONFIGURED)))
   define Build/Configure/rebuild
 	$(FIND) $(PKG_BUILD_DIR) -name \*.o -or -name \*.a | $(XARGS) rm -f
@@ -599,8 +596,6 @@ ifneq ($(wildcard $(PKG_BUILD_DIR)/.config_*),$(subst .configured_,.config_,$(ST
   endef
 endif
 
-EXTERNAL_PATCH_DIR:=$(TOPDIR)/openwrt-patches/package/network/services/hostapd/
-
 define Build/Configure
 	$(Build/Configure/rebuild)
 	$(if $(wildcard ./files/hostapd-$(CONFIG_VARIANT).config), \
@@ -613,7 +608,20 @@ endef
 
 define Build/InstallDev
 	$(INSTALL_DIR) $(1)/usr/include
+	$(INSTALL_DIR) $(1)/usr/lib/
+	$(CP) $(PKG_BUILD_DIR)/src/common/qca-vendor.h $(1)/usr/include
 	$(CP) $(PKG_BUILD_DIR)/src/drivers/nl80211_copy.h $(1)/usr/include/
+	$(CP) $(PKG_BUILD_DIR)/src/common/wpa_ctrl.h $(1)/usr/include
+	$(CP) $(PKG_BUILD_DIR)/src/utils/os.h $(1)/usr/include
+	$(CP) $(PKG_BUILD_DIR)/src/utils/common.h $(1)/usr/include
+	$(CP) $(PKG_BUILD_DIR)/src/utils/wpa_debug.h $(1)/usr/include
+ifneq ($(LOCAL_VARIANT),macsec)
+	ar rcs $(1)/usr/lib/libwpa_ctrl.a \
+		$(PKG_BUILD_DIR)/build/hostapd/src/common/wpa_ctrl.o \
+		$(PKG_BUILD_DIR)/build/hostapd/src/utils/os_unix.o \
+		$(PKG_BUILD_DIR)/build/hostapd/src/utils/wpa_debug.o \
+		$(PKG_BUILD_DIR)/build/hostapd/src/utils/common.o
+endif
 endef
 
 TARGET_CPPFLAGS := \
@@ -625,6 +633,12 @@ TARGET_CPPFLAGS := \
 	$(if $(CONFIG_WPA_MSG_MIN_PRIORITY),-DCONFIG_MSG_MIN_PRIORITY=$(CONFIG_WPA_MSG_MIN_PRIORITY))
 
 TARGET_LDFLAGS += -lubox -lubus -lblobmsg_json -lucode -lm -lnl-tiny -ludebug
+TARGET_CFLAGS += -fPIC
+
+ifdef CONFIG_DEBUG
+	TARGET_CFLAGS += -g3
+endif
+
 
 ifdef CONFIG_WPA_ENABLE_WEP
     DRIVER_MAKEOPTS += CONFIG_WEP=y
@@ -763,20 +777,11 @@ define Package/hostapd-common/install
 	  $(1)/etc/init.d $(1)/lib/netifd \
 	  $(1)/usr/share/acl.d \
 	  $(1)/usr/share/hostap
-ifeq (,$(findstring QSDK_Open,$(CONFIG_TARGET_PROFILE)))
 	$(INSTALL_BIN) ./files/dhcp-get-server.sh $(1)/lib/netifd/dhcp-get-server.sh
 	$(INSTALL_BIN) ./files/wpad.init $(1)/etc/init.d/wpad
 	$(INSTALL_BIN) ./files/wps-hotplug.sh $(1)/etc/rc.button/wps
 	$(INSTALL_DATA) ./files/wpad_acl.json $(1)/usr/share/acl.d
 	$(INSTALL_DATA) ./files/wpad.json $(1)/etc/capabilities
-endif
-ifneq (,$(findstring QSDK_Open,$(CONFIG_TARGET_PROFILE)))
-	$(INSTALL_BIN) $(FILES_DIR)/files/dhcp-get-server.sh $(1)/lib/netifd/dhcp-get-server.sh
-	$(INSTALL_BIN) $(FILES_DIR)/files/wpad.init $(1)/etc/init.d/wpad
-	$(INSTALL_BIN) $(FILES_DIR)/files/wps-hotplug.sh $(1)/etc/rc.button/wps
-	$(INSTALL_DATA) $(FILES_DIR)/files/wpad_acl.json $(1)/usr/share/acl.d
-	$(INSTALL_DATA) $(FILES_DIR)/files/wpad.json $(1)/etc/capabilities
-endif
 endef
 
 define Package/hostapd/install
@@ -791,14 +796,20 @@ Package/hostapd-mini/install = $(Package/hostapd/install)
 Package/hostapd-openssl/install = $(Package/hostapd/install)
 Package/hostapd-wolfssl/install = $(Package/hostapd/install)
 Package/hostapd-mbedtls/install = $(Package/hostapd/install)
-Package/hostapd-macsec/install = $(Package/hostapd/install)
+
+define Package/hostapd-macsec/install
+	$(call Install/hostapd,$(1))
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/hostapd/hostapd $(1)/usr/sbin/hostapd-macsec
+endef
 
 ifneq ($(LOCAL_TYPE),supplicant)
+ifneq ($(LOCAL_VARIANT),macsec)
   define Package/hostapd-utils/install
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/hostapd/hostapd_cli $(1)/usr/sbin/
   endef
 endif
+endif
 
 define Package/wpad/install
 	$(call Install/hostapd,$(1))
@@ -832,14 +843,20 @@ Package/wpa-supplicant-mbedtls/install = $(Package/wpa-supplicant/install)
 Package/wpa-supplicant-mesh-openssl/install = $(Package/wpa-supplicant/install)
 Package/wpa-supplicant-mesh-wolfssl/install = $(Package/wpa-supplicant/install)
 Package/wpa-supplicant-mesh-mbedtls/install = $(Package/wpa-supplicant/install)
-Package/wpa-supplicant-macsec/install = $(Package/wpa-supplicant/install)
+
+define Package/wpa-supplicant-macsec/install
+	$(call Install/supplicant,$(1))
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/wpa_supplicant/wpa_supplicant $(1)/usr/sbin/wpa_supplicant-macsec
+endef
 
 ifneq ($(LOCAL_TYPE),hostapd)
+ifneq ($(LOCAL_VARIANT),macsec)
   define Package/wpa-cli/install
 	$(INSTALL_DIR) $(1)/usr/sbin
 	$(CP) $(PKG_BUILD_DIR)/wpa_supplicant/wpa_cli $(1)/usr/sbin/
   endef
 endif
+endif
 
 ifeq ($(BUILD_VARIANT),supplicant-full-internal)
   define Package/eapol-test/install
-- 
2.34.1

